### PR TITLE
2364 notify user of missing mozAddonManager

### PR DESCRIPTION
**Pull request by fzzzy**

### CODE REVIEW

File: frontend/src/app/components/View.js [hunk 1]
```diff
@@ -92,12 +92,48 @@ export default class View extends React.Component {
...
+      return null;
+    }
+    if (!this.props.isFirefox) {
+      return null;
+    }
+    let title = "Something is wrong!";
+    let copy = "Something has gone wrong with Test Pilot. Please file a bug at https://github.com/mozilla/testpilot/issues/new and mention this error message.";
+    if (!this.props.isMinFirefox) {
+      title = "Upgrade firefox";
+      copy = "The version of firefox which is installed is too old. Please upgrade.";
```
Comment (pdehaan) [inline frontend/src/app/components/View.js]: Not sure how user-facing these error messages are, but these should probably be "Firefox" (same w/ "Upgrade firefox" above).

Comment (fzzzy) [reply frontend/src/app/components/View.js]: That's why this pr is a wip :) Some of these messages probably need translation too.

**Comment by lmorchard** [comment]
Can you add a component unit test for this message? Codecov is sad.

**Comment by fzzzy** [comment]
@lmorchard I have added test coverage and the coverage diff is now zero.

**Comment by fzzzy** [comment]
@johngruen Can you please take a look at this PR and decide if the strings I added need rewording?

**Comment by meandavejustice** [comment]
@fzzzy is localization planned for this?

File: frontend/src/app/components/View.js [hunk 2]
```diff
@@ -92,12 +92,48 @@ export default class View extends React.Component {
...
+    if (this.props.hasAddon === null) {
+      return null;
+    }
+    if (typeof navigator.mozAddonManager !== 'undefined') {
+      return null;
+    }
+    if (!this.props.isFirefox) {
+      return null;
+    }
+    let title = <h1 className="warning">Something is wrong!</h1>;
```
Comment (lmorchard) [inline frontend/src/app/components/View.js]: Yeah, we need some L10n for these strings

**Review by lmorchard — CHANGES_REQUESTED** [review]

**Review by lmorchard — APPROVED** [review]
Works for me! I do wish we had a bit more specific advice about what prefs are missing for dev environments, but just running through the quickstart from beginning to end resolved the warning for me on a fresh profile.

**Pull request merged**